### PR TITLE
Update channel service to reflect the new channel structure

### PR DESCRIPTION
**Pull request by JozefCeluch**
Updates the FIrebaseChannelService to reflect the updated structure of channels and their indexes.

### CODE REVIEW

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/service/FirebaseChannelService.java [hunk 1]
```diff
@@ -97,15 +95,107 @@ public void call() {
...
+                return Observable.create(new Observable.OnSubscribe<ChannelInfo>() {
+                    @Override
+                    public void call(final Subscriber<? super ChannelInfo> subscriber) {
+                        channelsDB.child(channelPath).addListenerForSingleValueEvent(new ValueEventListener() {
+                            @Override
+                            public void onDataChange(DataSnapshot dataSnapshot) {
+                                if (dataSnapshot.hasChildren()) {
+                                    subscriber.onNext(toChannelInfo(dataSnapshot));
+                                }
+                                subscriber.onCompleted();
```
Comment (takecare) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/service/FirebaseChannelService.java]: so deep! can't we extract some of this stuff?

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/service/FirebaseChannelService.java [hunk 2]
```diff
@@ -1,11 +1,14 @@
 package com.novoda.bonfire.channel.service;
 
+import android.support.annotation.NonNull;
```
Comment (takecare) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/service/FirebaseChannelService.java]: sure to add this dependency here?

Comment (JozefCeluch) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/service/FirebaseChannelService.java]: this class is in the app module

**Comment by JozefCeluch** [comment]
@takecare your comments have been addressed

**Comment by joetimmins** [comment]
Much more readable now without the nested Observables. I think it's always going to be a lot of Rx processing stuff going on here, because the data structure itself is very convoluted.

👍

**Pull request merged**